app/src/main/java/ca/klapstein/baudit/Data/CareProvider.java

```diff
@@ -1,10 +1,66 @@
 package ca.klapstein.baudit.Data;
 
+import java.util.Iterator;
+
+import ca.klapstein.baudit.Activities.LogoutDialog;
+import ca.klapstein.baudit.Managers.BauditRemoteManager;
+
 /**
  * Class that represents a Care Provider.
  */
 public class CareProvider extends User {
     private static final String TAG = "CareProvider";
 
     private PatientTreeSet assignedPatientTreeSet;
+
```

Comment:
many of these methods are already given by using a TreeSet.

Unless you are adding extra safety in getting/setting elements within the TreeSet I would not implement these methods.

Reply:
removed delete and add methods